lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -71,7 +72,9 @@ def date_range(self, start: AnyStr, end: AnyStr) -> list:
         start_date = date_parser.parse(start)
         end_date = date_parser.parse(end)
         assert start_date <= end_date
-        for m in rrule.rrule(rrule.MONTHLY, dtstart=start_date, until=end_date):
+        first_month = start_date.replace(day=1)
+        last_month = end_date + relativedelta(day=31)
```

Comment:
Why add 31 days instead of 1 month?

Reply:
And, if what we're gunning for is the last day, it seems that something like this would be appropos:
```
        first_month = start_date
        last_month = end_date.replace(day=1) + relativedelta(months=1) - relativedelta(days=1)
```

Reply:
And, why are you using `relativedelta()` with `day` (_singular_) instead of `days` (plural) or instead of using `replace()` as you did for `first_month`?

Reply:
> year, month, day, hour, minute, second, microsecond:

    Absolute information (argument is singular); adding or subtracting a
    relativedelta with absolute information does not perform an arithmetic
    operation, but rather REPLACES the corresponding value in the
    original datetime with the value(s) in relativedelta.

So the point here is to replace the day-of-month in the end `datetime` with the last day of that month. Using `days` would advance to the next month, without more complicated math like your suggestion, which we don't want. (And, yes, the parameter is maximized against the actual length of the month, so we'll get February 28 or 29, not 31, and there are relatively few months with more than 31 days ...)

Reply:
FYI, this change is copied and pasted from the production code equivalent. It was carefully researched and tested then, and probably still works. 😁